lenskit-core/src/main/java/org/lenskit/data/ratings/Rating.java

```diff
@@ -143,6 +169,51 @@ public double getValue() {
         return value;
     }
 
+    @Override
```

Comment:
Maybe we could create a abstract viewentity that implements hasAttribute etc. using reflection, maybe cacheing the results of reflection inspection (at a static level) for speedups.